### PR TITLE
cb153 Register every trace on trace mock

### PR DESCRIPTION
**Pull request by Brandonz07**
Every button with a flag should produce a trace on the “trace mock”, including: “Rechazar” “anular” …

image

### CODE REVIEW

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx [hunk 1]
```diff
@@ -28,10 +36,19 @@ export interface IRequirementsProps {
...
+  );
   const [showAprovalsModal, setShowAprovalsModal] = useState(false);
   const [isApproved, setIsApproved] = useState(false);
   const [showFlagMessage, setShowFlagMessage] = useState(false);
+  const [flagMessage, setFlagMessage] = useState({
+    title: "",
+    description: "",
+    appearance: "success" as "success" | "danger",
+  });
+  const { id } = useParams();
```
Comment (Andresbl123) [inline src/pages/board/outlets/financialReporting/Requirements/index.tsx:50]: Esto se puede pasar mejor como props, mejor desde el padre es decir desde `<FinancialReporting />`

**Review by Andresbl123 — APPROVED** [review]
cumple requerimiento

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx [hunk 2]
```diff
@@ -44,14 +60,68 @@ export const Requirements = (props: IRequirementsProps) => {
...
+      appearance: "success" | "danger";
+    }) => void,
+    setShowFlagMessage: (state: boolean) => void,
+    setShowApprovalstModal: (state: boolean) => void
+  ) => {
+    const justificationText = formData.textarea;
+
+    if (justificationText && id) {
+      const trace = {
+        trace_id: crypto.randomUUID(),
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/Requirements/index.tsx:79]: The trace_id should not be given by the frontend; this is a backend work. Therefore, the trace_id should be created automatically by the mock and not generated by the trace registration

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx [hunk 3]
```diff
@@ -44,14 +60,68 @@ export const Requirements = (props: IRequirementsProps) => {
...
+        trace_value: "Document approved",
+        credit_request_id: id,
+        use_case: "document_upload",
+        user_id: user,
+        execution_date: new Date().toISOString(),
+        justification: justificationText,
+        decision_taken_by_user: "approved",
+        trace_type: "novelty_document",
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/Requirements/index.tsx:87]: The trace_type is executed_task

Include the enumerator of the trace_type into the services/types
trace_type: novelty, executed_task, novelty_document, document, message

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx [hunk 4]
```diff
@@ -44,14 +60,68 @@ export const Requirements = (props: IRequirementsProps) => {
...
+        read_novelty: "N",
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/Requirements/index.tsx:88]: The read_novelty only applies to novelties (data generated by the client, as a message). Due to that, the read_novelty should be a void string ("") in this case.

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx [hunk 5]
```diff
@@ -44,14 +60,68 @@ export const Requirements = (props: IRequirementsProps) => {
...
+      };
+
+      const handleSuccess = () => {
+        setFlagMessage({
+          title: "Exito",
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/Requirements/index.tsx:93]: This should be in a config file

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx [hunk 6]
```diff
@@ -44,14 +60,68 @@ export const Requirements = (props: IRequirementsProps) => {
...
+          description: "La aprobación se ha completado correctamente.",
+          appearance: "success",
+        });
+        setShowFlagMessage(true);
+        setShowAprovalsModal(false);
+      };
+
+      const handleError = (error: Error) => {
+        setFlagMessage({
+          title: "Aprobación Fallida",
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/Requirements/index.tsx:103]: It should be defined in a config file.

File: src/pages/board/outlets/financialReporting/config.tsx [hunk 7]
```diff
@@ -96,18 +98,50 @@ export const handleConfirmCancel = (
...
+      trace_id: crypto.randomUUID(),
+      trace_value: "Document cancelled",
+      credit_request_id: id,
+      use_case: "document_cancel",
+      user_id: user,
+      execution_date: new Date().toISOString(),
+      justification: justificationText,
+      decision_taken_by_user: "cancelled",
+      trace_type: "novelty_document",
+      read_novelty: "N",
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/config.tsx:114]: The read_novelty only applies to novelties (data generated by the client, as a message). Due to that, the read_novelty should be a void string ("") in this case.

File: src/pages/board/outlets/financialReporting/config.tsx [hunk 8]
```diff
@@ -96,18 +98,50 @@ export const handleConfirmCancel = (
...
+    };
+
+    const handleSuccess = () => {
+      setFlagMessage({
+        title: "Anulación Confirmada",
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/config.tsx:119]: Relocate this information into a config file

**Review by carin91 — CHANGES_REQUESTED** [review]

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx (hunk 2 above)
Comment (Brandonz07) [reply src/pages/board/outlets/financialReporting/Requirements/index.tsx]: the trace_id is generated by the mock

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx (hunk 3 above)
Comment (Brandonz07) [reply src/pages/board/outlets/financialReporting/Requirements/index.tsx]: change trace_type

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx (hunk 4 above)
Comment (Brandonz07) [reply src/pages/board/outlets/financialReporting/Requirements/index.tsx]: change read_novelty

File: src/pages/board/outlets/financialReporting/Requirements/index.tsx (hunk 5 above)
Comment (Brandonz07) [reply src/pages/board/outlets/financialReporting/Requirements/index.tsx]: i moved to config

File: src/pages/board/outlets/financialReporting/config.tsx (hunk 8 above)
Comment (Brandonz07) [reply src/pages/board/outlets/financialReporting/config.tsx]: It's already in config

**Review by carin91 — APPROVED** [review]

**Pull request merged**